pvacseq/lib/vaccine_design.py

```diff
@@ -158,6 +159,221 @@ def anneal(self):
+
+#draw arc for amino acid inserts to junctions
+def draw_arc_junct(peptide, length, t, conversion_factor, junct_seq_space, circle_radius):
+    #t.pencolor("black")
+    t.circle(circle_radius, (conversion_factor * length) / 2)
+    t.pu()
+    reset = t.heading()
+    t.left(90)
+    t.back(junct_seq_space)
+    t.write(peptide, align="center")
+    t.forward(junct_seq_space)
+    t.setheading(reset)
+    t.pd()
+    t.circle(circle_radius, (conversion_factor * length) / 2)
+
+#print turtle screen to a postscript file, convert to pdf
+def output_screen(t, out_f):
+    ps_file = "/".join((out_f, "vaccine.ps"))
+    out_file = "/".join((out_f, "vaccine.jpg"))
+    ts = t.getscreen()
+    ts.getcanvas().postscript(file=ps_file)
+    os.system('convert -density 300 -quality 100 ' + ps_file + " " + out_file)
+    os.system('rm ' + ps_file)
+    return()
+
+def output_vaccine_png(input_file, out_f):
+    min_pep_length = 8
+    max_pep_length = 100
+
+    pep_seqs = []
+    with open(input_file, 'r') as input_f:
+        header = input_f.readline().strip()
+        for line in input_f:
+            pep_seqs.append(line.strip())
+    #remove >, get peptide names and junction scores from FASTA input
+    edited_header = header[1:]
+    fields = edited_header.split("|")
+    pep_ids_joined = fields[0].split(",")
+    pep_ids = []
+    for pep_id in pep_ids_joined:
+        if "." in pep_id:
+            mt, gene, var = pep_id.split(".")
+            pep_id = "-".join((gene,var))
+        pep_ids.append(pep_id)
+    junct_scores = fields[3].split(":")
+    junct_scores = junct_scores[1].split(",")
+
+    #Error if not a peptide sequence for every peptide ID
+    if len(pep_ids) != len(pep_seqs):
+        print("Error: Not an equal number of peptide sequences and peptide IDs")
+        sys.exit()
+
+    conversion_factor = get_conversion_factor(pep_seqs)
+
+    #determine number of peptides (not including junction additions)
+    num_peptides = 0
+    for pep in pep_seqs:
+        length = len(pep)
+        if length > 8 and length < 25:
+            num_peptides += 1
+            
+
+    #draw vaccine
+    turtle.setup(750,600)
+    t = turtle.Turtle()
+    myWin = turtle.Screen()
+    turtle.colormode(255)
+    turtle.mode("logo")
+    t.speed(0)
+    t.hideturtle()
+
+    #negative radius draws circle clockwise
+    circle_radius = -200
+    circle_pos = (-200,0)
+    header_pos = (0,0)
+    wht_space_angle = 15
+    pen_thick = 5
+    pen_thin = 2
+    pep_id_space = 45 + num_peptides
+    junct_seq_space = 25
+
+    draw_header(t, header_pos)
+    t.pu()
+    t.setpos(circle_pos)
+    t.pd()
+    t.pensize(pen_thick)
+    angle_parsed = 0
+    #add white space in circle before genes
+    draw_wht_space(t, circle_radius, wht_space_angle)
+    angle_parsed += wht_space_angle
+    draw_junction(t, pen_thin)
+
+    #draw main part of circle
+    junctions_parsed = 0
+    peptides_parsed = 0
+    for pep in pep_seqs:
+        pep_length = len(pep)
```

Comment:
The content of this for-loop body could also be pulled into a `draw_peptide` subroutine.